test/e2e/denylist_test.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2019 The prometheus-operator Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func testDenylist(t *testing.T) {
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup(t)
+
+	operatorNamespace := ctx.CreateNamespace(t, framework.KubeClient)
+	allowedNamespaces := ctx.CreateNamespace(t, framework.KubeClient)
+	deniedNamespaces := ctx.CreateNamespace(t, framework.KubeClient)
```

Comment:
Won't this be just one namespace? As the var name implies its more than one namespace.

Would it make sense to have more than one denied/allowed namespace in the e2e tests? Or should this test suffice and the rest is covered by unit tests anyways?

Reply:
Sounds good, let's add two denied and allowed namespaces. I have that case covered in unit tests, but it definitely makes sense to expand e2e on this.